src/meta/client/MetaClient.cpp

```diff
@@ -39,8 +41,22 @@ MetaClient::~MetaClient() {
 }
 
 
-void MetaClient::init() {
-    loadDataThreadFunc();
+bool MetaClient::waitForMetadReady() {
+    while (!ready_) {
```

Comment:
The caller has no way to abort this retry loop. This is not appropriate.

Reply:
Now, the code has changed! Please review! Thanks!